apps/web-giddh/src/app/subscription/payment-method-dialog/payment-method-dialog.component.ts

```diff
@@ -0,0 +1,224 @@
+    public isFormSubmitted: boolean = false;
+    /** Instance for payment method form */
+    public paymentMethodForm: FormGroup;
+    /** Hold payment provider list */
+    public paymentProviderList: any[] = [];
+    /** Hold selected payment method label form */
+    public paymentProvideLabel: string = "";
+    /** This will use for open window */
+    private openedWindow: Window | null = null;
+    /** Hold payment method list*/
+    public paymentMethodList: any[] = [];
+    /** True if api is in progress */
+    public isLoading: boolean = false;
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public inputData,
+        private changeDetection: ChangeDetectorRef,
+        public dialog: MatDialog,
+        private formBuilder: FormBuilder,
+        private router: Router,
+        private componentStore: PaymentMethodDialogComponentStore,
+        private generalService: GeneralService,
+        public dialogRef: MatDialogRef<any>
+    ) {
+    }
+
+    /**
+     * Hook cycle for component initialization.
+     *
+     * @memberof PaymentMethodDialogComponent
+     */
+    public ngOnInit(): void {
+        document.body?.classList?.add("subscription-sidebar");
+        this.dialogRef.updatePosition({ top: '0px', right: '0px' });
+        this.localeData = this.inputData?.localeData;
+        this.commonLocaleData = this.inputData?.commonLocaleData;
+        this.paymentProviderList = [{ label: this.localeData?.gocardless, value: "GOCARDLESS" }]
+        this.initForm();
+        this.getPaymentMethods();
+        this.saveProviderSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.redirectLink) {
+                this.openWindow(response.redirectLink);
+            }
+        });
+
+        this.deletePaymentSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.getPaymentMethods();
+            }
+        });
+
+        this.setDetaultPaymentMethodIsSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.getPaymentMethods();
+            }
+        });
+
+        window.addEventListener('message', event => {
+            if (this.router.url === '/pages/user-details/subscription') {
+                if (event?.data && typeof event?.data === "string" && event?.data === "GOCARDLESS") {
+                    this.isLoading = true;
+                    this.resetForm();
+                    this.getPaymentMethods();
+                    this.changeDetection.detectChanges();
+                }
+            }
+        });
+
+        this.paymentMethodList$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                setTimeout(() => {
+                    this.isLoading = false;
+                    this.paymentMethodList = response;
+                    this.changeDetection.detectChanges();
+                }, 1000);
```

Comment:
1 sec delay is too much, try to decrease it.

Reply:
Decreased to 700